pkg/sql/opt/exec/factory.go

```diff
@@ -326,11 +326,34 @@ type Factory interface {
 		table cat.Table,
 		insertCols ColumnOrdinalSet,
 		returnCols ColumnOrdinalSet,
-		checks CheckOrdinalSet,
+		checkCols CheckOrdinalSet,
 		allowAutoCommit bool,
 		skipFKChecks bool,
 	) (Node, error)
 
+	// ConstructInsertFastPath creates a node that implements a special (but very
+	// common) case of insert, satisfying the following conditions:
+	//  - the input is Values with at most InsertFastPathMaxRows, and there are no
```

Comment:
Whats stopping us from doing this in a batch by batch basis on the results from a subquery? It seems like this could work (we wouldn't save as much on the deduplication of lookup requests though)